lightning/src/ln/channelmanager.rs

```diff
@@ -2722,6 +2729,43 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		}
 	}
 
+	/// Send a payment that is probing the given route for liquidity. We calculate the
+	/// [`PaymentHash`] of probes based on a static secret and a random [`PaymentId`], which allows
+	/// us to easily discern them from real payments. This can be checked by calling
+	/// [`payment_is_probe`].
+	///
+	/// [`payment_is_probe`]: Self::payment_is_probe
+	pub fn send_probe_payment(&self, route: &Route) -> Result<(PaymentHash, PaymentId), PaymentSendFailure> {
+		let payment_id = PaymentId(self.keys_manager.get_secure_random_bytes());
+
+		let mut preimage = [0u8; 32];
+		preimage[..16].copy_from_slice(&self.probing_cookie_secret[..16]);
+		preimage[16..].copy_from_slice(&payment_id.0[..16]);
```

Comment:
Ah, you're right. I somehow thought the pre-image would have to be 32 bytes, but since I'm not using `PaymentPreimage` anyways this is not the case. Increased to 64 bytes now with https://github.com/lightningdevkit/rust-lightning/pull/1567/commits/22ac083f798476c185b4e9e925084c9684e8dde0.